src/cs/usfca/edu/histfavcheckout/controller/HistFavCheckoutHandler.java

```diff
@@ -73,6 +74,33 @@ public User addUser(PrimaryKey id) {
 		return ResponseEntity.status(HttpStatus.OK).body(product.get());
 	}
 	
+	public ResponseEntity<?> getTopFavs(int start, int num) {
+		int threshold = 10;
+		Optional<Product> products[] = productRepository.findAll();
+		if(products.length == 0) {
+			return ResponseEntity.status(HttpStatus.NOT_FOUND).body("No movie is added yet.");
+		}
+		int arrayLength = num;
+		if(num > threshold) {arrayLength = threshold;}
+		Optional<Product> topFavs[] = Arrays.copyOfRange(products, start, start + arrayLength);
+		int i = 0;
+		Product[] movies;
+		for(Optional<Product> product : topFavs) {
+			movies[i] = products[i++].get();
+		}
+		
+		for(int i = 0; i < movies.length-1; i++) {
```

Comment:
You're sorting by Faves here [after already selecting N products ](https://github.com/usfcs-perf-eng-s20/heroku-project-favs/pull/1/files#diff-0fae860ae0c1e35be827d26bc884c318R85). So the top Faves that you return isn't checking the whole data.

A better approach would be to sort the whole product data by decreasing order of numberOfFaves, then get the top N movies after. 

A SQL Query can help achieve this easily. You can use the query
`SELECT p FROM Product p ORDER BY p.numberOfFavorites DESC LIMIT :limit` 

So, in the [ProductRepository class](https://github.com/usfcs-perf-eng-s20/heroku-project-favs/blob/master/src/cs/usfca/edu/histfavcheckout/model/ProductRepository.java), you can add the following code:

`SELECT p FROM Product p ORDER BY p.numberOfFavorites DESC LIMIT :threshold` 
`public List<Product> findTopNFavoritedMovies(@Param("threshold") int threshold);`

Once you add that code, in the HistFavCheckoutHandler.java class you can call and instance of ProductRepository.findTopNFavoritedMovies(int threshold) instead of [findAll() that you currently call](https://github.com/usfcs-perf-eng-s20/heroku-project-favs/pull/1/files#diff-0fae860ae0c1e35be827d26bc884c318R79) and it will return the Top favorited movies up to the threshold you passed in the method.

You can see an example of this with the getUser API. See how the [findUserWithUserId](https://github.com/usfcs-perf-eng-s20/heroku-project-favs/blob/master/src/cs/usfca/edu/histfavcheckout/model/UserRepository.java#L19) function is called in the [HistFavCheckoutHandler.java class](https://github.com/usfcs-perf-eng-s20/heroku-project-favs/blob/master/src/cs/usfca/edu/histfavcheckout/controller/HistFavCheckoutHandler.java#L41)